Cython/Build/Inline.py

```diff
@@ -109,19 +109,175 @@ def _get_build_extension():
+            hashlib.sha1(_unicode(cache_key).encode('utf-8')).hexdigest()
+
+    if module_name in sys.modules:
+        # fast-path: extension was previously compiled and imported
+        #            return reference to existing module
+        module = sys.modules[module_name]
+        cython_inline_raw.cache[cache_key] = module
+        return module
+
+    # extension was not yet imported in this session
+    # construct the file-path for the python extension to load
+    build_extension = None
+    if cython_inline_raw.so_ext is None:
+        # Figure out and cache current extension suffix
+        build_extension = _get_build_extension()
+        cython_inline_raw.so_ext = build_extension.get_ext_filename('')
+    module_path = os.path.join(
+        module_cache_dir, module_name + cython_inline_raw.so_ext)
+    if not os.path.exists(module_cache_dir):
+        os.makedirs(module_cache_dir)
+
+    # check if a python extension still needs to be compiled
+    if force or not os.path.isfile(module_path):
+        if transform_code_fn:
+            # perform any code transformations if required
+            code, c_include_dirs, distutils_extra_compile_args = transform_code_fn(
+                    code, transform_code_args, list(c_include_dirs),
+                    list(distutils_extra_compile_args))
+        pyx_file = os.path.join(module_cache_dir, module_name + '.pyx')
+        fh = open(pyx_file, 'w')
+        try:
+            fh.write(code)
+        finally:
+            fh.close()
+        extension = Extension(
+            name=module_name,
+            sources=[pyx_file],
+            include_dirs=c_include_dirs,
+            extra_compile_args=distutils_extra_compile_args)
+        if build_extension is None:
+            build_extension = _get_build_extension()
+        build_extension.extensions = cythonize(
+            [extension],
+            include_path=cythonize_include_dirs,
+            compiler_directives=cythonize_compiler_directives,
+            **cythonize_kwargs)
+        build_extension.build_temp = os.path.dirname(pyx_file)
+        build_extension.build_lib = module_cache_dir
+        build_extension.run()
+
+    # load compiled python module
+    module = imp.load_dynamic(module_name, module_path)
+    cython_inline_raw.cache[cache_key] = module
+    return module
+
+# Cached values used by cython_inline_raw above.
+cython_inline_raw.so_ext = None
+cython_inline_raw.cache = {}
+
+
+def cython_inline_module(
+        code,
+        module_cache_dir=os.path.join(get_cython_cache_dir(), 'inline'),
+        c_include_dirs=None,
+        distutils_extra_compile_args=None,
+        cythonize_include_dirs=None,
+        cythonize_compiler_directives=None,
+        cythonize_kwargs=None,
+        language_level=None,
+        quiet=False,
+        force=False):
+
+    def preprocessing(code):
+        code = to_unicode(code)
+        code, literals = strip_string_literals(code)
```

Comment:
In this case, this is only done to be able to strip common indent. Perhaps make that clearer?